internal/security/bootstrapper/tcp/client.go

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright 2021 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package tcp
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+const (
+	dialTimeoutDuration = 5 * time.Second
+)
+
+// DialTcp will instantiate a new TCP dialer trying to connect to the TCP server specified by host and port
+func DialTcp(host string, port int, lc logger.LoggingClient) error {
+	tcpHost := strings.TrimSpace(host)
+	if len(tcpHost) == 0 || port == 0 {
+		// log and ignore those
+		lc.Warn(fmt.Sprintf("Skipping: no TCP server specified, host=%s, port=%d", host, port))
```

Comment:
Why not an error?

Reply:
need to look into this case.

Reply:
Based on the Golang's net.dial godoc, I've changed this to error when port is <= 0. Empty host string is allowed.